cornell-dti/idol
# Pass auth token in header instead of using session

### Summary 

This PR implements my proposal in [this comment](https://github.com/cornell-dti/idol/pull/124#pullrequestreview-655028464) to replace session with auth token in header. The firebase auth token verification should be efficient enough so that we don't need to use session to cache things.

It should auto resolve the data racing issue. I verified that by unchaining shoutout page requests and things can run without auto-sign-out issues.

### Test Plan 

- Visit every page to ensure that the data can be loaded.
- Open chrome dev tools to check that the auth token is passed in header
